python/tk_maya/panel_generation.py

```diff
@@ -8,10 +8,55 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
+import maya.cmds as cmds
+import maya.mel as mel
+import maya.utils
+
 from . import panel_util
 
 
-def dock_panel(engine, shotgun_panel, title, new_panel):
+# Prefix prepended to the Shotgun app panel unique identifier to create
+# the name given to the Qt widget at the root of the Shotgun app panel.
+SHOTGUN_APP_PANEL_PREFIX = "panel_"
```

Comment:
Hum, somewhat overkill when the full workspace control name is considered, for example `maya_panel_tk_multi_shotgunpanel_main`. Besides, this is an internal name never seen by the user.